inc/render/class-live-search-block.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * Live Search variant.
+ *
+ * @package ThemeIsle\GutenbergBlocks\Render
+ */
+
+namespace ThemeIsle\GutenbergBlocks\Render;
+
+/**
+ * Class Live_Search
+ */
+class Live_Search {
+	/**
+	 * The main instance var.
+	 *
+	 * @var Live_Search
+	 */
+	public static $instance = null;
+
+	/**
+	 * Initialize the class
+	 */
+	public function init() {
+		add_filter( 'render_block', array( $this, 'render_blocks' ), 10, 3 );
+	}
+
+	/**
+	 * Block render function for server-side.
+	 *
+	 * @param string $block_content Blocks content.
+	 * @param array  $block Blocks data.
+	 * @return mixed|string
+	 */
+	public function render_blocks( $block_content, $block ) {
+		if ( is_admin() || 'core/search' !== $block['blockName'] || ! isset( $block['attrs']['isLive'] ) ) {
+			return $block_content;
+		}
+
+		$asset_file = include OTTER_BLOCKS_PATH . '/build/blocks/live-search.asset.php';
+		wp_enqueue_script(
+			'otter-live-search',
+			OTTER_BLOCKS_URL . 'build/blocks/live-search.js',
+			$asset_file['dependencies'],
+			$asset_file['version'],
+			true
+		);
+		wp_localize_script(
+			'otter-live-search',
+			'liveSearchData',
+			array(
+				'nonce'   => wp_create_nonce( 'wp_rest' ),
+				'restUrl' => esc_url_raw( get_rest_url() ) . 'otter/v1/live-search',
+				'strings' => array(
+					/* translators: This is followed by the search string */
+					'noResults' => __( 'No results for', 'otter-blocks' ),
+				),
+			)
+		);
```

Comment:
I recommend adding to the typescript definition file for the window for better checking 🙏 .

In this line, https://github.com/Codeinwp/otter-blocks/blob/b7e0fee76d2114a7cae5693902ebf00616bfb6fe/src/blocks/frontend/live-search/index.ts#L24

It was not clear at first where that value came from. Also, this way, you miss some potential error checking from typescript 🐛